test/integration/isolated/assertion_helpers.go

```diff
@@ -0,0 +1,45 @@
+package isolated
+
+import (
+	"errors"
+	"io"
+	"os"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/test"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/integration/consts"
+)
+
+func requireNoResponseUDP(t *testing.T, udpGatewayURL string) {
+	t.Helper()
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		// For UDP lack of response (a timeout) means that we can't reach a service.
+		err := test.EchoResponds(test.ProtocolUDP, udpGatewayURL, "irrelevant")
+		assert.True(c, os.IsTimeout(err), "unexpected error: %v", err)
+	}, consts.IngressWait, consts.WaitTick)
```

Comment:
I'd consider exposing those wait and interval times in function signature.

Additionally you could consider mentioning in the name of the function that this will use `Eventually`-flavoured funcs underneath. WDYT?

Reply:
:+1: for naming change

for exposing those time related parameters, tbh they are good enough in most cases, so idk whether we should caller bother about setting them

Reply:
But if you insist I can make those params explicit

Reply:
> But if you insist I can make those params explicit

I believe those are fine for now. Just a thought to keep in mind that this is an are that might be improved IMHO.